tests/test_io.py

```diff
@@ -254,3 +255,13 @@ def test_io_datapackage(test_df, tmpdir):
     # read from csv assert that IamDataFrame instances are equal
     import_df = read_datapackage(file)
     assert_iamframe_equal(test_df, import_df)
+
+def test_io_netcdf(test_df, tmpdir):
+    # add column to `meta` and write to datapackage
+    test_df.set_meta(["a", "b"], "string")
+    
+    # read from csv assert that IamDataFrame instances are equal
```

Comment:
Thanks for the information! When calling `test_df` to compare with dataframe from `read_netcdf`, an AttributeError occurs as below. As `test_df` is usually passed to functions such as `pyam.to_csv(file)`, I think another `else` clause is needed to get data directly from `test_df` for comparison in this case.
![error_test_df](https://github.com/user-attachments/assets/e126fd7a-fca6-48a4-81f8-5ef383f2502b)

Reply:
@danielhuppmann thanks for the comment! I've updated the test but there is still one problem when comparing the meta data. The error message says `AssertionError: DataFrame.iloc[:, 0] (column name="string") are different`, and more specifically  `nan != nan`.

Since np.nan is different from np.nan by default, I am not sure how to solve this problem. Could you help me with this please?